Lesson18/main.js

```diff
@@ -0,0 +1,213 @@
+
+const createErrorMessage = (error) => {
+  const errorMessage = createElementWithClassName("p", "error-message");
+  errorMessage.textContent = error;
+  return errorMessage;
+}
+
+const fetchErrorHandling = async (response) => {
+  if (response.ok) {
+    return await response.json();
+  } else {
+    throw new Error("サーバーエラーが発生しました");
+  }
+}
+
+const getJsonOrError = async (url) => {
+  const response = await fetch(url);
+  const json = await fetchErrorHandling(response);
+  return json;
+}
+
+const getSlideImgData = async () => {
+  return new Promise((resolve) => {
+    setTimeout(() => resolve(getJsonOrError("https://myjson.dit.upm.es/api/bins/gzcn")), 3000);
+  })
+}
+
+const tryGetSlideImgData = async () => {
+  slideWrapper.appendChild(createLoading());
+  try {
+    const slideImgData = await getSlideImgData();
+    if (slideImgData.length === 0) {
+      throw new Error("No data...");
+    }
+    return slideImgData;
+  } catch (e) {
+    slideWrapper.appendChild(createErrorMessage(e));
+  } finally {
+    removeLoading();
+  }
+}
+
+const init = async () => {
+  const imgData = await tryGetSlideImgData();
+  if (imgData) {
+    initSlideItem(imgData);
+    initArrowButtons();
+    initIndicator(imgData);
+    slideWrapper.appendChild(createCounterOfSlide(imgData));
+  }
+}
+
+init();
+
+const initSlideItem = (imgSources) => {
+  for (let i = 0; i < imgSources.length; i++) {
+    const createdSlideItem = createSlideItem(imgSources[i]);
+    /**
+     * Set the is-displaying class to the first element by default 
+     */
+    i === 0 && createdSlideItem.classList.add("is-displaying");
+    slideList.appendChild(createdSlideItem);
+  }
+}
+
+const initArrowButtons = () => {
+  const createdArrowButtons = createArrowButtons();
+  slideWrapper.appendChild(createdArrowButtons);
+  setClickEventInArrowButton();
+}
+
+const initIndicator = (data) => {
+  slideWrapper.appendChild(createIndicator(data));
+  setClickEventInIndicator();
+}
+
+const createSlideItem = ({ image }) => {
+  const slideItem = createElementWithClassName("li", "slide-item");
+  const slideImg = createElementWithClassName("img", "slide-img");
+  slideImg.src = image;
+  slideItem.appendChild(slideImg);
+  return slideItem;
+}
+
+const createArrowButtons = () => {
+  const arrowBtnWrapper = createElementWithClassName("div", "arrowBtn-wrapper");
+  const arrowDirections = ["previous", "next"];
+  arrowDirections.forEach((arrowDirection) => {
+    const button = createElementWithClassName("button", `arrowBtn --${arrowDirection}`);
+    button.id = `js-${arrowDirection}Btn`;
+    button.value = arrowDirection;
+    /**
+     * Set previous attribute to disabled by default.
+     */
+    button.value === "previous" && button.setAttribute("disabled", true);
+    arrowBtnWrapper.appendChild(button).appendChild(document.createElement("span"));
+  });
+  return arrowBtnWrapper;
+}
+
+const createIndicator = (data) => {
```

Comment:
```suggestion
const createIndicator = (imageLength) => {
```

Reply:
The fix was made along with the above.
Please check.

57959a2